js/services/contact_service.js

```diff
@@ -76,10 +76,13 @@ angular.module('contactsApp')
 		}
 	};
 
-	this.create = function(newContact, addressBook) {
+	this.create = function(newContact, addressBook, uid) {
 		addressBook = addressBook || AddressBookService.getDefaultAddressBook();
 		newContact = newContact || new Contact(addressBook);
 		var newUid = uuid4.generate();
```

Comment:
Please put this into an else clause. We don't have to do this if we have a valid uid.